packages/react-devtools-shared/src/devtools/views/Profiler/ClearProfilingDataButton.js

```diff
@@ -8,22 +8,41 @@
  */
 
 import * as React from 'react';
-import {useCallback, useContext} from 'react';
+import {useContext} from 'react';
 import {ProfilerContext} from './ProfilerContext';
 import Button from '../Button';
 import ButtonIcon from '../ButtonIcon';
 import {StoreContext} from '../context';
+import {SchedulingProfilerContext} from 'react-devtools-scheduling-profiler/src/SchedulingProfilerContext';
 
 export default function ClearProfilingDataButton() {
   const store = useContext(StoreContext);
-  const {didRecordCommits, isProfiling} = useContext(ProfilerContext);
+  const {didRecordCommits, isProfiling, selectedTabID} = useContext(
+    ProfilerContext,
+  );
+  const {clearSchedulingProfilerData, schedulingProfilerData} = useContext(
+    SchedulingProfilerContext,
+  );
   const {profilerStore} = store;
 
-  const clear = useCallback(() => profilerStore.clear(), [profilerStore]);
+  let doesHaveData = false;
+  if (selectedTabID === 'scheduling-profiler') {
+    doesHaveData = schedulingProfilerData !== null;
+  } else {
+    doesHaveData = didRecordCommits;
+  }
+
+  const clear = () => {
+    if (selectedTabID === 'scheduling-profiler') {
+      clearSchedulingProfilerData();
```

Comment:
could you share a bit more context on this change? what's the difference between clearing the scheduling profiler data vs clearing the profiler store?

the profiler store corresponds to the current profiler tab vs the new scheduling profiler tab?

Reply:
Yes. There are two profilers, each with a separate data source. One is React-generated and the other comes from the browser's Performance profile JSON. This kind of sucks, and I'd like to smooth over this distinction in the future (see #22015).